### PR TITLE
Add Admin Select fixes #8

### LINKED ISSUE #8: Add Admin Select
- [x] add Admin Select ( consider react-select instead of materail ui select )
- [x] add Logout option

### PR DESCRIPTION
**Pull request by wiola-dreszler**

### CODE REVIEW

File: client/src/App/Header/Header.tsx [hunk 1]
```diff
@@ -45,6 +51,8 @@ const Header: React.FC = () => {
                         <MenuDesktop
                             loadingState={loadingState}
                             items={menuItems}
+                            currentUser={currentUser}
+                            onLogoutCb={onLogoutCb}
```
Comment (wojciech-bilicki) [inline client/src/App/Header/Header.tsx]: Don't use abbreviations like `Cb`. If you want to have full name please change it to `callback`, but in my opinion `onLogout` as a name is sufficient.

File: client/src/App/Header/Header.tsx [hunk 2]
```diff
@@ -72,7 +80,12 @@ const Header: React.FC = () => {
                         setDrawerOn(false);
                     }}
                 >
-                    <MenuMobile loadingState={loadingState} items={menuItems} />
+                    <MenuMobile
+                        loadingState={loadingState}
+                        items={menuItems}
+                        currentUser={currentUser}
+                        onLogoutCb={onLogoutCb}
```
Comment (wojciech-bilicki) [inline client/src/App/Header/Header.tsx]: ditto

File: client/src/App/Header/MenuDesktop.tsx [hunk 3]
```diff
@@ -30,7 +38,7 @@ const MenuDesktop: React.FC<MenuDesktopProps> = ({ loadingState, items }) => (
                 <LoadingBox state={loadingState} />
             </Grid>
             <Grid item>
-                <UserSelect />
+                <UserSelect currentUser={currentUser} onLogoutCb={onLogoutCb} />
```
Comment (wojciech-bilicki) [inline client/src/App/Header/MenuDesktop.tsx]: So here's an interesting thought... you're passing `onLogoutCb` few levels deep into the component tree. How about storing this in a Context?

File: client/src/App/Header/header.hooks.ts [hunk 4]
```diff
@@ -82,19 +91,29 @@ const useConnectionState = (): boolean | null => {
     return connected;
 };
 
-const useMenuItems = (dataset: Dataset | null): Array<NavItem> => {
+const useMenuItems = (
+    dataset: Dataset | null,
+    currentUser: UserInfo | null,
+): Array<NavItem> => {
     // Calculate extra page for menu
     let menuItems: Array<NavItem> = [];
```
Comment (wojciech-bilicki) [inline client/src/App/Header/header.hooks.ts]: why is this let and not a const?

File: client/src/Auth/authenticatedRoute.tsx [hunk 5]
```diff
@@ -13,3 +13,11 @@ export function withAuthentication(matcher: Matcher<{}, Context>) {
         context.currentUser ? matcher : redirect('/auth'),
     );
 }
+
+export function withAdminContentProtection(matcher: Matcher<{}, Context>) {
+    return map((request: NaviRequest<Context>, context: Context) =>
+        context.currentUser && context.currentUser.is_admin
```
Comment (wojciech-bilicki) [inline client/src/Auth/authenticatedRoute.tsx:17]: why do we have `is_admin` and not `isAdmin`? Shouldn't everything be camelCased?

**Review by wojciech-bilicki — CHANGES_REQUESTED** [review]

**Review by maciejdrozdowski — APPROVED** [review]

**Review by wojciech-bilicki — APPROVED** [review]

**Pull request merged**